Validation of a statement for renaming a table fails with ParseException: Cannot parse statement: Encountered unexpected token: "to" "TO"

My code:
```
	@Test
	public void testJSQParser()
	{
		String sql = "alter table public.table2 rename to table3;";
		Validation validation = new Validation(List.of(DatabaseType.POSTGRESQL), sql);
		List<ValidationError> errors = validation.validate();
		Assert.assertTrue(errors.toString(), errors.isEmpty());
	}
```
Result:
```
java.lang.AssertionError: [ValidationError [
statement=alter table public.table2 rename to table3;
capability=parsing
errors=[ParseException: Cannot parse statement: Encountered unexpected token: "to" "TO"
    at line 1, column 34.

Was expecting:

    "COLUMN"
]
]]
```
Same with Oracle. This statement can be executed in both.

--- a/build.gradle
+++ b/build.gradle
@@ -12,7 +12,7 @@ plugins {
 }
 
 group = 'com.github.jsqlparser'
-version = '4.2-SNAPSHOT'
+version = '4.3-SNAPSHOT'
 description = 'JSQLParser library'
 java.sourceCompatibility = JavaVersion.VERSION_1_8
 

--- a/src/main/java/net/sf/jsqlparser/statement/alter/Alter.java
+++ b/src/main/java/net/sf/jsqlparser/statement/alter/Alter.java
@@ -70,6 +70,11 @@ public class Alter implements Statement {
         if (useOnly) {
             b.append("ONLY ");
         }
+
+        if (alterExpressions.size()>0 && alterExpressions.get(0).getOperation()==AlterOperation.RENAME_TABLE && alterExpressions.get(0).isUsingIfExists()) {
+            b.append("IF EXISTS ");
+        }
+
         b.append(table.getFullyQualifiedName()).append(" ");
 
         Iterator<AlterExpression> altIter = alterExpressions.iterator();

--- a/src/main/java/net/sf/jsqlparser/statement/alter/AlterExpression.java
+++ b/src/main/java/net/sf/jsqlparser/statement/alter/AlterExpression.java
@@ -32,6 +32,7 @@ public class AlterExpression {
 
   private AlterOperation operation;
   private String optionalSpecifier;
+  private String newTableName;
   private String columnName;
   private String columnOldName;
   // private ColDataType dataType;
@@ -246,6 +247,14 @@ public class AlterExpression {
     this.fkSourceColumns = fkSourceColumns;
   }
 
+  public String getNewTableName() {
+    return newTableName;
+  }
+
+  public void setNewTableName(String newTableName) {
+    this.newTableName = newTableName;
+  }
+
   public String getColumnName() {
     return columnName;
   }
@@ -367,6 +376,8 @@ public class AlterExpression {
     
     if (operation== AlterOperation.UNSPECIFIC) {
         b.append(optionalSpecifier);
+    } else if (operation== AlterOperation.RENAME_TABLE) {
+      b.append("RENAME TO ").append(newTableName);
     } else {
         b.append(operation).append(" ");
 

--- a/src/main/java/net/sf/jsqlparser/statement/alter/AlterExpression.java
+++ b/src/main/java/net/sf/jsqlparser/statement/alter/AlterExpression.java
@@ -45,7 +45,7 @@ public class AlterExpression {
   private String ukName;
   private Index index = null;
   private String constraintName;
-  private boolean constraintIfExists;
+  private boolean usingIfExists;
 
   private Set<ReferentialAction> referentialActions = new LinkedHashSet<>(2);
 
@@ -289,12 +289,12 @@ public class AlterExpression {
     this.constraintName = constraintName;
   }
 
-  public boolean isConstraintIfExists() {
-    return constraintIfExists;
+  public boolean isUsingIfExists() {
+    return usingIfExists;
   }
 
-  public void setConstraintIfExists(boolean constraintIfExists) {
-    this.constraintIfExists = constraintIfExists;
+  public void setUsingIfExists(boolean usingIfExists) {
+    this.usingIfExists = usingIfExists;
   }
 
   public List<String> getPkColumns() {
@@ -377,6 +377,7 @@ public class AlterExpression {
     if (operation== AlterOperation.UNSPECIFIC) {
         b.append(optionalSpecifier);
     } else if (operation== AlterOperation.RENAME_TABLE) {
+
       b.append("RENAME TO ").append(newTableName);
     } else {
         b.append(operation).append(" ");
@@ -428,7 +429,7 @@ public class AlterExpression {
           }
         } else if (constraintName != null) {
           b.append("CONSTRAINT ");
-          if (constraintIfExists) {
+          if (usingIfExists) {
             b.append("IF EXISTS ");
           }
           b.append(constraintName);
@@ -515,8 +516,8 @@ public class AlterExpression {
     return this;
   }
 
-  public AlterExpression constraintIfExists(boolean constraintIfExists) {
-    this.setConstraintIfExists(constraintIfExists);
+  public AlterExpression withUsingIfExists(boolean usingIfExists) {
+    this.setUsingIfExists(usingIfExists);
     return this;
   }
 

--- a/src/main/java/net/sf/jsqlparser/statement/alter/AlterOperation.java
+++ b/src/main/java/net/sf/jsqlparser/statement/alter/AlterOperation.java
@@ -10,5 +10,5 @@
 package net.sf.jsqlparser.statement.alter;
 
 public enum AlterOperation {
-    ADD, ALTER, DROP, MODIFY, CHANGE, ALGORITHM, RENAME, COMMENT, UNSPECIFIC;
+    ADD, ALTER, DROP, MODIFY, CHANGE, ALGORITHM, RENAME, RENAME_TABLE, COMMENT, UNSPECIFIC;
 }

--- a/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
+++ b/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
@@ -5596,7 +5596,7 @@ AlterExpression AlterExpression():
                        }
                )
                |
-         (  <K_CONSTRAINT> [<K_IF> <K_EXISTS> { alterExp.setConstraintIfExists(true); } ]
+         (  <K_CONSTRAINT> [<K_IF> <K_EXISTS> { alterExp.setUsingIfExists(true); } ]
          (tk=<S_IDENTIFIER>    | tk=<S_QUOTED_IDENTIFIER>)
              {
                 alterExp.setConstraintName(tk.image);
@@ -5655,13 +5655,21 @@ Alter AlterTable():
     Alter alter = new Alter();
     Table table;
     AlterExpression alterExp;
-
+    boolean usingIfExists = false;
 }
 {
-    <K_ALTER> <K_TABLE> [ <K_ONLY> { alter.setUseOnly(true); } ] table=Table() { alter.setTable(table); }
-     ( alterExp=AlterExpression() { alter.addAlterExpression(alterExp); }
-       ("," alterExp=AlterExpression() { alter.addAlterExpression(alterExp); } )*
-     )
+    <K_ALTER> <K_TABLE>
+    [ <K_ONLY> { alter.setUseOnly(true); } ]
+    [ LOOKAHEAD(2) <K_IF> <K_EXISTS> { usingIfExists = true; } ]
+
+    table=Table() { alter.setTable(table); }
+
+    alterExp=AlterExpression() { if (usingIfExists)
+                                    alter.addAlterExpression( alterExp.withUsingIfExists(true) );
+                                  else
+                                    alter.addAlterExpression(alterExp); }
+
+    ("," alterExp=AlterExpression() { alter.addAlterExpression(alterExp); } )*
 
     {
         return alter;

--- a/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
+++ b/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
@@ -4973,7 +4973,7 @@ CreateTable CreateTable():
     // see https://docs.oracle.com/cd/B19306_01/server.102/b14200/statements_7002.htm#i2126725
     // table properties , all these are optional
     [ rowMovement = RowMovement() { createTable.setRowMovement(rowMovement); }]
-    [ <K_AS> select = SelectWithWithItems( ) { createTable.setSelect(select, fa
+    [ <K_AS> select = SelectWithWithItems( ) { createTable.setSelect(select, false); }]
     [
     <K_LIKE> ( LOOKAHEAD("(" Table() ")") "(" likeTable=Table() { createTable.setLikeTable(likeTable, true); } ")"
                  | likeTable=Table() { createTable.setLikeTable(likeTable, false); } )

--- a/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
+++ b/src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt
@@ -5613,13 +5613,14 @@ AlterExpression AlterExpression():
          sk3 = RelObjectName() {alterExp.addParameters(sk3); }
       )
       |
-      (<K_RENAME> {alterExp.setOperation(AlterOperation.RENAME);}
-        (
-          <K_COLUMN> {alterExp.hasColumn(true);}
-          (tk=<S_IDENTIFIER>    | tk=<S_QUOTED_IDENTIFIER>) { alterExp.setColOldName(tk.image);}
-           (<K_TO>)
-          (tk2=<S_IDENTIFIER>    | tk2=<S_QUOTED_IDENTIFIER>) { alterExp.setColumnName(tk2.image);}
-        )
+      LOOKAHEAD(2)  <K_RENAME> <K_COLUMN> { alterExp.setOperation(AlterOperation.RENAME); alterExp.hasColumn(true);}
+                    ( tk=<S_IDENTIFIER>    | tk=<S_QUOTED_IDENTIFIER> ) { alterExp.setColOldName(tk.image); }
+                    <K_TO>
+                    (tk2=<S_IDENTIFIER>    | tk2=<S_QUOTED_IDENTIFIER>) { alterExp.setColumnName(tk2.image); }
+      |
+      (
+        <K_RENAME> <K_TO> {alterExp.setOperation(AlterOperation.RENAME_TABLE);}
+        (tk2=<S_IDENTIFIER>    | tk2=<S_QUOTED_IDENTIFIER>) { alterExp.setNewTableName(tk2.image);}
       )
       |
       (<K_COMMENT> {alterExp.setOperation(AlterOperation.COMMENT);}

--- a/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
@@ -104,6 +104,15 @@ public class RenameTableStatementTest {
 
       // this needs to succeed
       ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
+
+      sqlStr = "ALTER TABLE IF EXISTS public.oldTableName RENAME TO newTableName";
+      TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
+
+      //  should fail when not supported in Oracle
+      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
+
+      // this needs to succeed
+      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
   }
   
   @Test

--- a/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
@@ -10,7 +10,6 @@
 
 package net.sf.jsqlparser.statement.alter;
 
-import java.util.List;
 import net.sf.jsqlparser.JSQLParserException;
 import net.sf.jsqlparser.parser.CCJSqlParserUtil;
 import net.sf.jsqlparser.schema.Table;
@@ -22,6 +21,8 @@ import net.sf.jsqlparser.util.validation.ValidationTestAsserts;
 import net.sf.jsqlparser.util.validation.feature.DatabaseType;
 import org.junit.Assert;
 import org.junit.Test;
+
+import java.util.List;
 
 /**
  *
@@ -90,31 +91,31 @@ public class RenameTableStatementTest {
     String sqlStr = "RENAME oldTableName TO newTableName";
     TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
 
-      // this needs to succeed
-      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
+    // this needs to succeed
+    ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
 
-      // this should fail when not supported in Postgres
-      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
+    // this should fail when not supported in Postgres
+    ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
 
-      sqlStr = "ALTER TABLE public.oldTableName RENAME TO newTableName";
-      TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
+    sqlStr = "ALTER TABLE public.oldTableName RENAME TO newTableName";
+    TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
 
-      //  should fail when not supported in Oracle
-      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
+    // this needs to succeed according to: https://docs.oracle.com/cd/B28359_01/server.111/b28286/statements_3001.htm
+    ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
 
-      // this needs to succeed
-      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
+    // this needs to succeed
+    ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
 
-      sqlStr = "ALTER TABLE IF EXISTS public.oldTableName RENAME TO newTableName";
-      TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
+    sqlStr = "ALTER TABLE IF EXISTS public.oldTableName RENAME TO newTableName";
+    TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
 
-      //  should fail when not supported in Oracle
-      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
+    //  should fail when IF EXISTS is not supported in Oracle 11
+    ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
 
-      // this needs to succeed
-      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
+    // this needs to succeed
+    ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
   }
-  
+
   @Test
   public void testObjectAccess() {
       Table oldTable = new Table("oldTableName");

--- a/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
+++ b/src/test/java/net/sf/jsqlparser/statement/alter/RenameTableStatementTest.java
@@ -88,8 +88,22 @@ public class RenameTableStatementTest {
   @Test
   public void testValidator() throws JSQLParserException {
     String sqlStr = "RENAME oldTableName TO newTableName";
+    TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
 
-    ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
+      // this needs to succeed
+      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
+
+      // this should fail when not supported in Postgres
+      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
+
+      sqlStr = "ALTER TABLE public.oldTableName RENAME TO newTableName";
+      TestUtils.assertSqlCanBeParsedAndDeparsed(sqlStr, true);
+
+      //  should fail when not supported in Oracle
+      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.ORACLE);
+
+      // this needs to succeed
+      ValidationTestAsserts.validateNoErrors(sqlStr, 1, DatabaseType.POSTGRESQL);
   }
   
   @Test

